ang/civicase/activity/actions/services/move-copy-activity-action.service.js

```diff
@@ -96,58 +96,26 @@
      * @returns {Array} api call configuration
      */
     function prepareApiCalls (activitiesObject, operation, model) {
-      if (activitiesObject.selectedActivities.length === 1) {
-        return prepareAPICallsForSingleActivity(activitiesObject, operation, model);
-      } else {
-        return prepareAPICallsForMultipleActivities(activitiesObject, operation, model);
-      }
-    }
-
-    /**
-     * Prepare the API calls for the move/copy operation
-     *
-     * @param {object} activitiesObject object containing configuration of activities
-     * @param {string} operation move or copy operation
-     * @param {object} model model object for dialog box
-     * @returns {Array} api call configuration
-     */
-    function prepareAPICallsForSingleActivity (activitiesObject, operation, model) {
-      var activity = activitiesObject.selectedActivities[0];
-
-      if (operation === 'copy') {
-        delete activity.id;
-      }
-
-      activity.subject = model.subject;
-      activity.case_id = model.case_id;
-
-      return [['Activity', 'create', activity]];
-    }
-
-    /**
-     * Prepare the API calls for the move/copy operation
-     *
-     * @param {object} activitiesObject object containing configuration of activities
-     * @param {string} operation move or copy operation
-     * @param {object} model model object for dialog box
-     * @returns {Array} api call configuration
-     */
-    function prepareAPICallsForMultipleActivities (activitiesObject, operation, model) {
       var action = operation === 'copy' ? 'copybyquery' : 'movebyquery';
+      var selectedActivitiesIds = _.map(activitiesObject.selectedActivities, 'id');
+      var isSingleActivity = selectedActivitiesIds.length === 1;
 
       if (activitiesObject.isSelectAll) {
         return [['Activity', action, {
           params: activitiesObject.searchParams,
           case_id: model.case_id
         }]];
-      } else {
+      } else if (isSingleActivity) {
         return [['Activity', action, {
           case_id: model.case_id,
-          id: activitiesObject.selectedActivities.map(function (activity) {
-            return activity.id;
-          }),
+          id: selectedActivitiesIds,
           subject: model.subject
         }]];
+      } else {
+        return [['Activity', action, {
+          case_id: model.case_id,
+          id: selectedActivitiesIds
+        }]];
```

Comment:
This can be refactored as 
```javascript
var apiCallParams = { case_id: model.case_id };

if (activitiesObject.isSelectAll) {
  apiCallParams.params = activitiesObject.searchParams;
} else {
  if (isSingleActivity) {
    apiCallParams.subject = model.subject;
  }

  apiCallParams.id = selectedActivitiesIds;
}

return [['Activity', action, apiCallParams]];
```

Reply:
looks better!